pyramid/session.py

```diff
@@ -128,7 +129,24 @@ def check_csrf_token(request,
 
     .. versionadded:: 1.4a2
     """
-    supplied_token = request.params.get(token, request.headers.get(header, ""))
+    # We want to start out with an empty string token which will never match
+    # any valid CSRF token, so that if we fail to locate a CSRF token then
+    # CSRF checks will fail.
+    supplied_token = ""
+
+    # If this is a POST request, then we'll check the POST body to see if it
+    # has a token. We explicitly use request.POST here because CSRF tokens
+    # should never appear in an URL as doing so is a security issue. We also
+    # explicitly check for request.POST here as we do not support sending form
+    # encoded data over anything but a request.POST.
+    if request.method == "POST":
```

Comment:
I had forgotten that was changed in WebOb, it used to do it unconditionally, but it was fixed because it would replace it with a FakeCGIBody, even on non form data.